hera_cal/lstbin.py

```diff
@@ -552,70 +547,52 @@ def config_lst_bin_files(data_files, dlst=None, atol=1e-10, lst_start=None, lst_
     time_arrays = []
     for di, dfs in enumerate(data_files):
         # get times
-        dlsts, dtimes, larrs, tarrs = io.get_file_times(dfs, filetype='uvh5')
-
-        # get lmin: LST of first integration from first file
-        if di == 0:
-            lmin = larrs[0][0]
-
-        # unwrap relative to lmin
-        for la in larrs:
-            if la[0] < lmin:
-                la += 2 * np.pi
-
-        # get lmax
-        if di == (len(data_files) - 1):
-            lmax = larrs[-1][-1]
-
+        _, _, larrs, tarrs = io.get_file_times(dfs, filetype='uvh5')
         # append
         lst_arrays.append(larrs)
         time_arrays.append(tarrs)
 
-    # get starting LST for output binning
+    # get begin_lst from lst_start or from the first JD in the data_files
     if lst_start is None:
-        lst_start = lmin
-
-    # if lst_start is sufficiently below lmin, shift everything down an octave
-    if lst_start < (lmin - np.pi):
-        lst_arrays = [[la - 2 * np.pi for la in day] for day in lst_arrays]
-        lmin -= 2 * np.pi
-        lmax -= 2 * np.pi
-
-    # get beginning LST for lst_grid
-    if fixed_lst_start:
-        begin_lst = lst_start
-    else:
-        begin_lst = lmin
-
-    # get stopping LST for output binning
-    if lst_stop is None:
-        lst_stop = lmax
-    if lst_stop < begin_lst:
-        lst_stop += 2 * np.pi
+        all_lsts = [lst for larrs in lst_arrays for larr in larrs for lst in larr]
+        all_times = [time for tarrs in time_arrays for tarr in tarrs for time in tarr]
+        lst_start = all_lsts[np.argmin(all_times)]
+    begin_lst = lst_start
 
     # make LST grid
     lst_grid = make_lst_grid(dlst, begin_lst=begin_lst, verbose=verbose)
     dlst = np.median(np.diff(lst_grid))
 
-    # get starting and stopping indicies
-    start_index = np.argmin(np.abs(lst_grid - lst_start))
-    stop_index = np.argmin(np.abs(lst_grid - lst_stop))
+    # enforce that lst_arrays are in the same range as the lst_grid
+    for larrs in lst_arrays:
+        for larr in larrs:
+            while np.any(larr < np.min(lst_grid)):
+                larr[larr < np.min(lst_grid)] += 2 * np.pi
+            while np.any(larr > np.max(lst_grid)):
+                larr[larr > np.max(lst_grid)] -= 2 * np.pi
 
     # get number of output files
-    nfiles = int(np.ceil(float(stop_index - start_index) / ntimes_per_file))
+    nfiles = int(np.ceil(1.0 * len(lst_grid) / ntimes_per_file))
```

Comment:
nitpicking but: you don't need the 1.0. int division results in float for python3+. if you want to make it python2 compatible I think the more readable thing is float(), rather than implicit type casting via float multiplication. but not really a big thing anyways

Reply:
fair point. fixed.